cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -691,27 +718,60 @@ fun Node.followNextDFGEdgesUntilHit(
                             // from).
                             // We try to pop from the stack and only select the elements with the
                             // matching index.
-                            indexStack?.checkAndPop(it.granularity as IndexedDataflowGranularity) ==
-                                true
+                            ctx.indexStack.checkAndPop(
+                                it.granularity as IndexedDataflowGranularity
+                            ) == true
```

Comment:
Is the comparison with `true` still needed?

Reply:
Probably not